.ci/common.sh

```diff
@@ -28,31 +30,41 @@ function should_run_job {
          ## Travis merges the PR commit into origin/master
          ## This identifies the PR's original commit
          ## if it notices a merge commit
-         local HEAD="HEAD"
```

Comment:
CIs are full of special git clones ..... does it make sense to make
```
if [[ TRAVIS ]]; then
  # do some actions/modifications specific to Travis
fi
``` 
https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
most likely all CIs set such variable, so some special clones will be handled separately, no need to try make general solutions

Reply:
We would need a variable for each CI as this `common.sh` touches on 3-4 different CIs. There was so many different variations that I can't remember specifics right now, but I think some CIs required similar overrides between master CI and PR CI.

My current thought right now was to fix bug in 1 CI and ensure it causes no issues in other CIs.

Reply:
ok, if we run into more problems we will try to use customization if possible.